lib/internal/fs.js

```diff
@@ -18,7 +19,7 @@ const {
 
 function assertEncoding(encoding) {
   if (encoding && !Buffer.isEncoding(encoding)) {
-    throw new Error(`Unknown encoding: ${encoding}`);
+    throw new errors.TypeError('ERR_INVALID_OPT_VALUE', 'encoding', encoding);
```

Comment:
I think I agree that sticking to the generic one makes sense, but open discussion  on the merits of a more specific one.